packages/admin-next/dashboard/src/routes/promotions/common/edit-rules/components/rules-form-field/rules-form-field.tsx

```diff
@@ -34,20 +26,84 @@ type RulesFormFieldType<TSchema extends FieldValues> = {
     | "application_method.target_rules"
 }
 
-export const RulesFormField = <TSchema extends FieldValues>({
+const generateRuleAttributes = (rules?: PromotionRuleResponse[]) =>
+  (rules || []).map((rule) => ({
+    id: rule.id,
+    required: rule.required,
+    field_type: rule.field_type,
+    disguised: rule.disguised,
+    attribute: rule.attribute!,
+    operator: rule.operator!,
+    values:
+      rule.field_type === "number"
+        ? rule.values
+        : rule?.values?.map((v: { value: string }) => v.value!),
+  }))
+
+export const RulesFormField = ({
   form,
   ruleType,
-  fields,
-  attributes,
-  operators,
-  removeRule,
-  updateRule,
-  appendRule,
   setRulesToRemove,
   rulesToRemove,
   scope = "rules",
-}: RulesFormFieldType<TSchema>) => {
+  promotionId,
+}: RulesFormFieldType) => {
   const { t } = useTranslation()
+  const formData = form.getValues()
+  const { attributes } = usePromotionRuleAttributes(ruleType, formData.type)
+  const { operators } = usePromotionRuleOperators()
+
+  const { fields, append, remove, update, replace } = useFieldArray({
+    control: form.control,
+    name: scope,
+    keyName: scope,
+  })
+
+  const watchType = useWatch({
```

Comment:
Can we rename this to something more obvious (it sounds like a function now), eg. `ruleTypeValue`